### ISSUE TITLE
New Dialog (modal) does not visibly show a close button if there is no message

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#11306]
### Issue Summary

While exploring the Dialog templates I have found that the close button does not show (it renders but is not visible) when there is no message declared in the Dialog.

From what I can see, this will only impact admin use for a small edge case. Models in workflow that do not have a latest revision, probably snippets could be set up this way or manual imports of Pages without setting a latest_revision.

This is visible in the Styleguide page though.

<img width="2053" alt="Screenshot 2023-12-05 at 8 38 30 pm" src="https://github.com/wagtail/wagtail/assets/1396140/0b539c4a-bd8b-4816-95e3-7ebca6aab9b4">

### Proposed solution

We could look at making the icon use the default colour (`theme('colors.icon-primary')`) and then override this when there is a message. 

1. Update the default colours in client/scss/components/_dialog.scss
2. Set a different colour either with Tailwind classes or a custom class in wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html

Something like this...

```html
        <div class="w-dialog__overlay" data-action="click->w-dialog#hide"></div>
        <div class="w-dialog__box">
            <button type="button" class="w-dialog__close-button{% if message_heading and message_icon_name %} w-text-grey-600{% endif %}" aria-label="{% trans 'Close dialog' %}" data-action="w-dialog#hide">
                {% icon name='cross' classname="w-dialog__close-icon" %}
            </button>
```

### Steps to Reproduce

1. In a new Wagtail project, install the Styleguide app
4. Navigate to the styleguide page via the settings menu - http://localhost:8000/admin/styleguide/
5. In the dialogs section, trigger the 'Simple dialog'
6. Expected: Close button should be visible
7. Actual: Close button is there (can be tabbed to) but is not visible

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (no)

### Technical details

-   Python version: 3.11
-   Wagtail version: 6.0a
-   Browser version: Firefox 120 on macOS 13.5

### Additional info

The close button for the same modal used for the site facing accessibility content shows a visible close button, this has differen styling though. See https://github.com/wagtail/wagtail/blob/main/client/scss/components/_userbar.scss#L235

<img width="2050" alt="Screenshot 2023-12-05 at 8 39 18 pm" src="https://github.com/wagtail/wagtail/assets/1396140/9bf40984-ca3d-481d-b157-7b3ef0f96692">

**Comment by lb-** [comment]
@nandini584  - first can you try to reproduce the issue with the steps provided, check you see it (maybe try in Chrome, I only tested in Firefox).

If you can reproduce this, add a comment with a screenshot.

From there, see how you go with a fix, we do not assign issues to anyone outside of the core team though.

https://docs.wagtail.org/en/stable/contributing/first_contribution_guide.html#how-can-i-be-assigned-an-issue-to-contribute-to

**Comment by nandini584** [comment]
Here I have reproduced the issue as you mentioned in the brave browser...

![Screenshot 2023-12-05 184448](https://github.com/wagtail/wagtail/assets/117419442/924717b2-cd31-4e46-b264-51fbd704b4f5)

**Comment by laymonage** [comment]
FYI @lb- I think in our design reviews we're planning to use the same styling as the one in the accessibility checker everywhere, so the proposed solution should probably be that. I'll check again with Ben E and let you know

**Comment by nandini584** [comment]
I have added a PR for this @lb-

**Comment by thibaudcolas** [comment]
Also reported by @albinazs as part of [WCAG 2.2 AAA* audit of the Wagtail admin – Nov 2023](https://github.com/wagtail/wagtail/discussions/11180).

In our dialog modals, _Close icon is almost invisible in dark theme_. This is failure of _1.4.11 Non-text Contrast_:
